Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransform.java

```diff
@@ -0,0 +1,579 @@
+     * @param direction
+     * @param _interface
+     * @param isRequesting
+     * @param webContextProperties
+     * @param serviceName
+     * @param response
+     * @return
+     */
+    public final LogEventRequestType transformMsgToAuditMsg(T request, AssertionType assertion,
+        NhinTargetSystemType target, String direction, String _interface, boolean isRequesting, Properties webContextProperties, String serviceName, K response) {
+        LOG.trace("Begin AuditDataTransform -> transformMsgToAuditMsg() ----");
+        if (request == null && response == null) {
+            LOG.error("message Object was null");
+            return null;
+        }
+        if (assertion == null) {
+            LOG.error("Assertion was null");
+            return null;
+        }
+
+        LogEventRequestType result = new LogEventRequestType();
+
+        AuditMessageType auditMsg = new AuditMessageType();
+        //****************************Construct Event Identification**************************
+
+        auditMsg.setEventIdentification(createEventIdentification(serviceName, isRequesting));
+
+        //*********************************Construct Active Participant************************
+        //Active Participant for human requester only required for requesting gateway
+        if (isRequesting) {
+            AuditMessageType.ActiveParticipant participantHumanFactor = getActiveParticipant(assertion.getUserInfo());
+            auditMsg.getActiveParticipant().add(participantHumanFactor);
+        }
+        AuditMessageType.ActiveParticipant participantSource = getActiveParticipantSource(isRequesting, webContextProperties, serviceName);
+        AuditMessageType.ActiveParticipant participantDestination = getActiveParticipantDestination(target, isRequesting, webContextProperties, serviceName);
+        auditMsg.getActiveParticipant().add(participantSource);
+        auditMsg.getActiveParticipant().add(participantDestination);
+
+        //******************************Constuct Participation Object Identification*************
+        // Assign ParticipationObjectIdentification
+        auditMsg = getParticipantObjectIdentification(request, response, assertion, auditMsg);
+
+        //Create the AuditSourceIdentifierType object
+        String communityId = getMessageCommunityId(assertion, target, isRequesting);
+        AuditSourceIdentificationType auditSource = getAuditSourceIdentificationType();
+
+        //******************************Constuct Audit Source Identification**********************
+        auditMsg.getAuditSourceIdentification().add(auditSource);
+
+        //Set the all the required data
+        result.setAuditMessage(auditMsg);
+        result.setDirection(direction);
+        result.setInterface(_interface);
+        //set the target community identifier
+        result.setCommunityId(communityId);
+
+        LOG.trace("End AuditDataTransform transformMsgToAuditMsg() ----");
+        return result;
+    }
+
+    /**
```

Comment:
As a rule of thumb, we want full JavaDoc comments everywhere we add / modify code, but it's **extremely** important for abstract methods in template classes -- this will make it easier if someone else ends up writing an implementation later.